### PR TITLE
review: test: fix AssertJ code generation

### PR DESCRIPTION
**Pull request by SirYwell**
The check to ensure the assertions are all up-to-date didn't run.
This PR fixes that, and also fixes the missing types. We also need concrete classes for interfaces that don't have one specific implementation, otherwise newly introduced code wouldn't compile anymore.

Additionally, interfaces are public now, otherwise methods returning those interfaces are basically useless.

### CODE REVIEW

**Comment by MartinWitt** [comment]
@SirYwell there are a lot of changed files. Can you summarize the change?

**Comment by SirYwell** [comment]
> @SirYwell there are a lot of changed files. Can you summarize the change?

I edited the description, I hope that helps.
